src/bootstrap/dist.rs

```diff
@@ -996,32 +1003,17 @@ impl Step for Src {
         let image = tmpdir(builder).join(format!("{}-image", name));
         let _ = fs::remove_dir_all(&image);
 
-        let dst = image.join("lib/rustlib/src");
-        let dst_src = dst.join("rust");
+        // A lot of tools expect the rust-src component to be entirely in this directory, so if you
+        // change that (e.g. by adding another directory `lib/rustlib/src/foo` or
+        // `lib/rustlib/src/rust/foo`), you will need to go around hunting for implicit assumptions
+        // and fix them...
+        let dst_src = image.join("lib/rustlib/src/rust");
         t!(fs::create_dir_all(&dst_src));
 
         let src_files = ["Cargo.lock"];
         // This is the reduced set of paths which will become the rust-src component
-        // (essentially libstd and all of its path dependencies)
-        let std_src_dirs = [
-            "src/build_helper",
-            "src/liballoc",
-            "src/libcore",
-            "src/libpanic_abort",
-            "src/libpanic_unwind",
-            "src/libstd",
-            "src/libunwind",
-            "src/libtest",
-            "src/libterm",
-            "src/libprofiler_builtins",
-            "src/stdarch",
-            "src/libproc_macro",
-            "src/tools/rustc-std-workspace-core",
-            "src/tools/rustc-std-workspace-alloc",
-            "src/tools/rustc-std-workspace-std",
-        ];
-
-        copy_src_dirs(builder, &std_src_dirs[..], &[], &dst_src);
+        // (essentially libstd and all of its path dependencies).
+        copy_src_dirs(builder, &builder.src, &["library"], &[], &dst_src);
```

Comment:
@Mark-Simulacrum Are you intentionally dropping `build_helper` here?

Reply:
Yes, AFAICT, it's not used by library crates at all.